### PR TITLE
55 add y block

### PR DESCRIPTION
**Pull request by asilenko**
Solved  Issue #55 Add "Y" block.
YBlock Class added with tests and javadoc.

### CODE REVIEW

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.epam.prejap.tetris.block;
+
+/**
+ * @author Anna Silenko
```
Comment (TJB4epam) [inline src/main/java/com/epam/prejap/tetris/block/YBlock.java:4]: Please, keep the JD order with the tags.

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ *Represents Y block in BlockFeed
+ * @see BlockFeed
+ *
```
Comment (TJB4epam) [inline src/main/java/com/epam/prejap/tetris/block/YBlock.java:8]: No need for this empty line, please cut, superfluous.

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            {1, 0, 1},
+            {0, 1, 0},
+            {0, 1, 0}
+    };
+
+    public YBlock() {
+        super(IMAGE);
+    }
+
+}
```
Comment (TJB4epam) [inline src/main/java/com/epam/prejap/tetris/block/YBlock.java:22]: All files should have an empty line at the end.

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        assertEquals(actual, expected, "Number of cols is incorrect. Expected: " + expected + ", actual: " + actual);
+    }
+
+    @Test(dataProvider = "dotsRepresentation")
+    public void dotAtReturnCorrectValue(int i, int j, int expected, String message) {
+        int actual = yBlock.dotAt(i, j);
+        assertEquals(actual, expected, message);
+    }
+
+    @DataProvider
```
Comment (TJB4epam) [inline src/test/java/com/epam/prejap/tetris/block/YBlockTest.java:29]: Cool. The only thing to improve would be to GROUP the tests.

Please, create a necessary testng.xml file with the config. See TestNG docs for an example file and info on groups.

All tests should belong to some groups.

**Review by TJB4epam — COMMENTED** [review]
Close to finish. Few small remarks, largest thing TBD is testng groups.

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java (hunk 3 above)
Comment (asilenko) [reply src/main/java/com/epam/prejap/tetris/block/YBlock.java:20]: Empty line added.

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java (hunk 2 above)
Comment (asilenko) [reply src/main/java/com/epam/prejap/tetris/block/YBlock.java:5]: Empty line cutted.

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java (hunk 4 above)
Comment (asilenko) [reply src/test/java/com/epam/prejap/tetris/block/YBlockTest.java:30]: Tests grouped.

**Comment by asilenko** [comment]
Ready for review.

**Review by TJB4epam — APPROVED** [review]
Small things, details, and it can go in.

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package com.epam.prejap.tetris.block;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+@Test(groups = {"blockShape"})
```
[the remark this answers is not in the document]
Comment (asilenko) [reply src/test/java/com/epam/prejap/tetris/block/YBlockTest.java]: Fixed

File: src/test/resources/testng.xml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="TetrisSuite" verbose="1">
+    <test name="TetrisTests">
+        <groups>
+            <define name="all">
+                <include name="blockShape"/>
```
[the remark this answers is not in the document]
Comment (asilenko) [reply src/test/resources/testng.xml]: Fixed

**Comment by asilenko** [comment]
Ready for review

File: src/main/java/com/epam/prejap/tetris/block/YBlock.java (hunk 1 above)
Comment (marcin-ogorzalek-EPAM) [reply src/main/java/com/epam/prejap/tetris/block/YBlock.java]: Done

**Comment by marcin-ogorzalek-EPAM** [comment]
@asilenko please resolve the merge conflits

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (marcin-ogorzalek-EPAM) [reply src/test/java/com/epam/prejap/tetris/block/YBlockTest.java]: Done

File: src/test/resources/testng.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (marcin-ogorzalek-EPAM) [reply src/test/resources/testng.xml]: Done

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java [hunk 7]
```diff
@@ -5,7 +5,7 @@
 
 import static org.testng.Assert.*;
 
-@Test(groups = {"blockShape"})
+@Test(groups = {"blockShapes", "blocks"})
```
Comment (marcin-ogorzalek-EPAM) [inline src/test/java/com/epam/prejap/tetris/block/YBlockTest.java:8]: Why do we have a second group `"blocks"` now? What is the reasoning behind this? I'm not saying it is wrong but I'm curious.

**Review by marcin-ogorzalek-EPAM — CHANGES_REQUESTED** [review]
I'm uncertain about `"blocks"` groupe. What's more, there are merge conflicts.

File: src/test/java/com/epam/prejap/tetris/block/YBlockTest.java (hunk 7 above)
Comment (asilenko) [reply src/test/java/com/epam/prejap/tetris/block/YBlockTest.java:8]: Tomasz recommended it https://github.com/EPAM-Online-Courses/tetris/pull/58#discussion_r649792266

Comment (marcin-ogorzalek-EPAM) [reply src/test/java/com/epam/prejap/tetris/block/YBlockTest.java:8]: Sorry I missed the last part of that comment.

**Review by marcin-ogorzalek-EPAM — APPROVED** [review]
Conflicts solved, doubts explained

**Comment by marcin-ogorzalek-EPAM** [comment]
Well, actually I have one doubt. @asilenko did you modified `TBlock` and tests for it. Content of the previously merged PR appears as something new here. It seems that there is a problem with history (if I am right).

**Comment by asilenko** [comment]
> Well, actually I have one doubt. @asilenko did you modified `TBlock` and tests for it. Content of the previously merged PR appears as something new here. It seems that there is a problem with history (if I am right).


I did not modify TBlock tests... I have some problem with history :-)  
TBlock has testing.xml outside Test directory. So now there is testing.xml and 'my' testng.xml in Test directory.

**Comment by marcin-ogorzalek-EPAM** [comment]
As already mentioned by Andrii `rebase` may be a key here.

**Comment by asilenko** [comment]
I had some trouble with using `git rebase` for the first time. It should be ok now.

File: src/test/java/com/epam/prejap/tetris/block/TBlockTest.java [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
 import static org.testng.Assert.assertEquals;
 
-@Test(groups = "blockShapes")
+@Test(groups = {"blockShapes", "blocks"})
```
Comment (TJB4epam) [inline src/test/java/com/epam/prejap/tetris/block/TBlockTest.java:8]: I will let that in, but... how is "blockShapes" group different from "blocks"?

**Review by TJB4epam — CHANGES_REQUESTED** [review]

**Comment by TJB4epam** [comment]
Too quick click! POM not bumped up, that's the reason why you don't have conflicts here. :-)

Please, bump up the version in a POM, after all, we're adding a new feature.

File: src/test/java/com/epam/prejap/tetris/block/TBlockTest.java (hunk 8 above)
Comment (asilenko) [reply src/test/java/com/epam/prejap/tetris/block/TBlockTest.java:8]: https://github.com/EPAM-Online-Courses/tetris/pull/58#discussion_r649792266
You recommended it?

**Comment by asilenko** [comment]
Probably ready :-)

File: src/test/java/com/epam/prejap/tetris/block/TBlockTest.java (hunk 8 above)
Comment (asilenko) [reply src/test/java/com/epam/prejap/tetris/block/TBlockTest.java:8]: I made some changes in testng.xml to make intentions behind creating both "blockShapes" and "blocks" more clear.

**Comment by asilenko** [comment]
@TJB4epam would you please take a look?

**Comment by asilenko** [comment]
Is this the proper solution, you're looking for @TJB4epam ?

**Comment by marcin-ogorzalek-EPAM** [comment]
@asilenko there is some change that worries me. 2 classes (other shapes) are deleted. Are we removing some functionalities here?

**Comment by asilenko** [comment]
> @asilenko there is some change that worries me. 2 classes (other shapes) are deleted. Are we removing some functionalities here?

One Test class with @Factory replaced those test classes, as recommended -> https://github.com/EPAM-Online-Courses/tetris/pull/84#discussion_r652426931
This is my first @Factory usage. Did I misunderstood it's concept of it?

**Comment by marcin-ogorzalek-EPAM** [comment]
@asilenko sorry for my mistake. I took those as removing functionalities (in this case tests) no replacing those with `@Factory`. This my previous comment is invalid.

File: pom.xml [hunk 9]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.epam.prejap</groupId>
     <artifactId>tetris</artifactId>
-    <version>0.4</version>
+    <version>0.5</version>
```
Comment (TJB4epam) [inline pom.xml:9]: TY!

**Review by TJB4epam — APPROVED** [review]
Nice! Much obliged!

**Pull request merged**